internal/oc/release.go

```diff
@@ -513,8 +652,62 @@ func getIcspContents(mirrorConfig []mirrorregistries.RegistriesConf) ([]byte, er
 	return contents, nil
 }
 
-func removeIcspFile(filename string) {
-	if filename != "" {
-		os.Remove(filename)
-	}
+// getFlagsFromHelp gets the names of the flags of the given command. To do so it removes all the flags from the
+// command, executes it with the '--help' flag and parses the output, assuming that it has the format used by the 'oc'
+// command. It returns an array with the names of the flags in alphabetical order.
+func getFlagsFromHelp(log logrus.FieldLogger, executer executer.Executer, command string) (result []string, err error) {
```

Comment:
I think I would like to see more tests around extraction of flags from help specifically, just so we can see that edge cases have been covered. 

I know that we potentially cover this in tests later though, so not sure if this is excessive.

This might be a case of testing this function and then non longer having to worry about edge cases in tests later.

Reply:
I added some more cases, please take a look again.